### PR TITLE
Fpl demos

### PR DESCRIPTION
**Pull request by clewis7**
- [ ] intro demo
     - [ ] just need to link in gallery and guide when merged
- [ ] neuro demo
     - [ ] need to add dataset to repo
- [ ] add slides to cluster

Probably shouldn't merge until we merge https://github.com/fastplotlib/fastplotlib/pull/511, https://github.com/fastplotlib/fastplotlib/pull/513, and https://github.com/fastplotlib/fastplotlib/pull/509 ...should happen sometime this week :D

### CODE REVIEW

File: docs/examples/05_fpl_intro.py [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+# to create the new data.
+
+# create new random dat of the same shape as the original image
+new_data = np.random.rand(*image_graphic.data.value.shape)
+
+# update the data
+image_graphic.data = new_data
+
+# %%
+# **Question:** Can you change the colormap of the image? See [here](https://matplotlib.org/stable/gallery/color/colormap_reference.html) for a list of colormaps.
```
Comment (kushalkolar) [inline docs/examples/05_fpl_intro.py:123]: This will change soon, we're going to use the cmap lib

File: docs/examples/05_fpl_intro.py [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+
+#add this as an animation function
+fig_v.add_animations(update_data)
+
+fig_v.show(sidecar=True)
+
+# %%
+# ### Adding another animation function {.strip-code,.keep-text}
+#
+# **Question:** Can you add an animation function that toggles the colormap of the image between "plasma" and "hsv"?
```
Comment (kushalkolar) [inline docs/examples/05_fpl_intro.py]: Not a very practical anim example?

Comment (kushalkolar) [reply docs/examples/05_fpl_intro.py]: Suggest something else, maybe on data?

Comment (clewis7) [reply docs/examples/05_fpl_intro.py]: we already have one with the data above, so I was trying to think of something else to do

**Comment by clewis7** [comment]
For tomorrow's practice, just going to run everything locally from our [fens repo](https://github.com/fastplotlib/fastplotlib-fens2024)

**Pull request closed without merging**